src/javascript/app/japan/cashier.js

```diff
@@ -11,16 +11,33 @@ const CashierJP = (() => {
     const onLoad = (action) => {
         if (jpClient() && !jpResidence()) BinaryPjax.loadPreviousUrl();
         const $container = $('#japan_cashier_container');
-        BinarySocket.wait('get_settings').then(() => {
-            $container.setVisibility(1);
-            if (action === 'deposit') {
-                $('#name_id').text(`${(Client.get('loginid') || 'JP12345')} ${(State.getResponse('get_settings.first_name') || 'Joe Bloggs')}`);
-            } else if (action === 'withdraw') {
-                $('#id123-control22598118').val(Client.get('loginid'));
-                $('#id123-control22598060').val(Client.get('email'));
-                $('#japan_cashier_container button').on('click', (e) => {
-                    const result = errorHandler();
-                    if (!result) e.preventDefault();
+        BinarySocket.send({ cashier_password: 1 }).then((response) => {
+            if (!response.error && response.cashier_password === 1) {
+                $container.find('#cashier_locked_message').setVisibility(1);
+            } else {
+                BinarySocket.send({ get_account_status: 1 }).then((response_status) => {
+                    if (!response_status.error && /cashier_locked/.test(response_status.get_account_status.status)) {
+                        $container.find('#cashier_locked_message').html('Your cashier is locked.').setVisibility(1); // Locked from BO
+                    } else {
+                        const limit = State.getResponse('get_limits.remainder');
+                        if (typeof limit !== 'undefined' && limit < 1) {
+                            $container.find('#cashier_locked_message').html('You have reached the limit.').setVisibility(1);
```

Comment:
same here. and do we need to be more specific about what limit they have reached? they will take care of it in translation but it's good for us too